flash/text/classification/data.py

```diff
@@ -27,54 +28,44 @@
 from flash.core.data.process import Deserializer, Postprocess, Preprocess
 from flash.core.integrations.labelstudio.data_source import LabelStudioTextClassificationDataSource
 from flash.core.utilities.imports import _TEXT_AVAILABLE, requires
+from flash.text.classification.tokenizers.base import BaseTokenizer
 
 if _TEXT_AVAILABLE:
     from datasets import Dataset, load_dataset
-    from transformers import AutoTokenizer, default_data_collator
+    from transformers import default_data_collator
     from transformers.modeling_outputs import SequenceClassifierOutput
 
+    from flash.text.classification.tokenizers import TEXT_CLASSIFIER_TOKENIZERS
+
 
 class TextDeserializer(Deserializer):
     @requires("text")
-    def __init__(self, backbone: str, max_length: int, use_fast: bool = True, **kwargs):
+    def __init__(self, tokenizer: BaseTokenizer):
         super().__init__()
-        self.backbone = backbone
-        self.tokenizer = AutoTokenizer.from_pretrained(backbone, use_fast=use_fast, **kwargs)
-        self.max_length = max_length
+
+        self.tokenizer = tokenizer
 
     def deserialize(self, text: str) -> Tensor:
-        return self.tokenizer(text, max_length=self.max_length, truncation=True, padding="max_length")
+        return self.tokenizer(text)
 
     @property
     def example_input(self) -> str:
         return "An example input"
 
-    def __getstate__(self):  # TODO: Find out why this is being pickled
-        state = self.__dict__.copy()
-        state.pop("tokenizer")
-        return state
-
-    def __setstate__(self, state):
-        self.__dict__.update(state)
-        self.tokenizer = AutoTokenizer.from_pretrained(self.backbone, use_fast=True)
-
```

Comment:
@ethanwharris is there a way to reproduce the error locally to test possible fixes? 

The previous approach would not work in this PR because the tokenizer can be now retrained from scratch. One solution could be to "decompose" and "recompose" the tokenizer in the get/set state methods.